client/me/security-2fa-sms-settings/index.jsx

```diff
@@ -220,46 +202,59 @@ const Security2faSMSSettings = createReactClass( {
 					<FormButtonsBar className="security-2fa-sms-settings__buttons">
 						<FormButton
 							disabled={ this.getSubmitDisabled() }
-							onClick={ function ( event ) {
+							onClick={ ( event ) => {
 								gaRecordEvent( 'Me', 'Clicked On 2fa Use App Button' );
 								this.onVerifyByApp( event );
-							}.bind( this ) }
+							} }
 						>
-							{ this.state.submittingForm ? savingLabel : this.props.translate( 'Verify via App' ) }
+							{ this.props.isUpdatingUserSettings
+								? savingLabel
+								: this.props.translate( 'Verify via App' ) }
 						</FormButton>
 
 						<FormButton
 							disabled={ this.getSubmitDisabled() }
 							isPrimary={ false }
-							onClick={ function ( event ) {
+							onClick={ ( event ) => {
 								gaRecordEvent( 'Me', 'Clicked On 2fa Use SMS Button' );
 								this.onVerifyBySMS( event );
-							}.bind( this ) }
+							} }
 						>
-							{ this.state.submittingForm ? savingLabel : this.props.translate( 'Verify via SMS' ) }
+							{ this.props.isUpdatingUserSettings
+								? savingLabel
+								: this.props.translate( 'Verify via SMS' ) }
 						</FormButton>
 
 						<FormButton
 							className="security-2fa-sms-settings__cancel-button"
 							isPrimary={ false }
-							onClick={ function ( event ) {
+							onClick={ ( event ) => {
 								gaRecordEvent( 'Me', 'Clicked On Step 1 2fa Cancel Button' );
 								this.props.onCancel( event );
-							}.bind( this ) }
+							} }
 						>
-							{ this.state.submittingForm ? savingLabel : this.props.translate( 'Cancel' ) }
+							{ this.props.isUpdatingUserSettings ? savingLabel : this.props.translate( 'Cancel' ) }
 						</FormButton>
 					</FormButtonsBar>
 				</form>
 			</div>
 		);
-	},
-} );
+	}
+}
 
-export default flowRight(
+export default compose(
 	protectForm,
 	localize,
-	connect( ( state ) => ( {
-		countriesList: getCountries( state, 'sms' ),
-	} ) )
+	connect(
+		( state ) => ( {
+			isUpdatingUserSettings: isUpdatingUserSettings( state ),
+			userSettings: getUserSettings( state ),
```

Comment:
Now that we're relying on the Redux user settings, we don't have a guarantee that data will be loaded. Should we have a `QueryUserSettings` in this component just in case?

If you agree on this, I have a follow-up question: should we block from rendering or render a placeholder while user settings are still loading?

Reply:
There will be a `<QueryUserSettings />` higher up in the tree added in #49002.

> should we block from rendering or render a placeholder while user settings are still loading

This also should be handled by `<TwoStep />` refactored in #49002.

Reply:
Agreed this is not ideal but I'd rather follow up and have the code of both PRs merged before making such additions 😬

Reply:
Makes sense 👍